showcase/src/app/shared/components/example-viewer/example-viewer.component.ts

```diff
@@ -22,13 +29,34 @@ export class ExampleViewerComponent implements OnInit {
 	public filesPath = "undefined";
 	@Input()
 	public exampleTitle = "undefined";
+	@Input()
+	public id?: string;
+
+	public get exampleId(): string {
+		return (
+			this.id ||
+			this.translate
+				.instant(this.exampleTitle)
```

Comment:
I don't think is a good idea to use the `exampleTitle` as a fallback in case there is no id, because there might be cases where the title can be the same for 2 or more viewers in the same page.

I suggest to just use the `id` and throw an error if it is not provided... what do you think?

Reply:
I agree that the title isn't the greatest id, but I wouldn't throw an error if the id is not set. I would rather generated a random id (`example-1823947`) or disable the functionality when there is no id set.

Reply:
Right, then I suggest we go for option 1: generate a random id.

If you want, once you implement the change, I can take over the PR and add the `id` input to every ExampleViewer in the showcase ;)  I just want to be sure we have meaningful/descriptive URLs of every example.

Reply:
I went for option 2 and disabled the anchor link when no id is given. I didn't realise but a random ID won't work. Because the same element will have a different id when visiting the page again, that kind of defeats the purpose I think.

I could make a hash of the title, this at least would not require the translate module being imported.

Reply:
True, random id's won't work. So I go back to my first idea, to throw an error if the id is not given, this will make sure we don't forget to set this id and the implementation will be simple so no Translate module required. To be honest, I don't find any reason why we should make this behaviour optional... it is just used in our showcase so we should be responsible in maintaining this as functional/complete as possible.

As I said, if you don't have time to adapt all the places where we use the `ExampleViewer`, I can do it in this PR once you implement this change.

Reply:
I personally don't like throwing errors for "non-essential" properties and forcing developers to use them (even if it is just us 😂). Maybe sometimes you would not want to add the anchor (you have two examples next to each other and add a title with an anchor or you just have a single example on the page). If we do want a reminder so we don't forget about it I would suggest a warning.
@SuperITMan, @nicanac what do you think I should do?

As for the current examples. I indeed think I won't have time to add them everywhere. 😕 But the PR obviously is still not finished. Because I can't seem to find how to handle landing on an url with a hash part. 😖